packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -754,6 +754,9 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
     return Rect.fromLTRB(topLeft.dx, topLeft.dy, bottomRight.dx, bottomRight.dy);
   }
 
+  /// Whether this focus node is already attached to the widget tree.
+  bool get hasAttachment => _context != null;
```

Comment:
```suggestion
  bool get attached => _attachment != null;
```

Reply:
_context may be non-null even if the attachment has been detached. It is probably a bug we should fix.

Reply:
Ahh, OK 😅